### PR TITLE
adding more tools options in distro_tools.py

### PR DESCRIPTION
**Pull request by vaishnavibhat**
adding lsvio, lsdevinfo, usysident, usysattn, ofpathname
to the list. Changing the directory of yaml file

Signed-off-by: Vaishnavi Bhat <vaishnavi@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by vaishnavibhat** [comment]
# avocado run distro_tools.py -m distro_tools.yaml 
/usr/local/lib/python3.6/site-packages/avocado_framework-75.1-py3.6.egg/avocado/plugins/run.py:214: FutureWarning: The following arguments will be changed to boolean soon: sysinfo, output-check, failfast, keep-tmp and ignore-missing-references.
  FutureWarning)
JOB ID     : 0558a9fc2d919a4f20ab0dca95335c8610472ea6
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2020-02-17T12.14-0558a9f/job.log
 (01/26) distro_tools.py:DisrtoTool.test;lsslot_pci-8c15: SKIP: supported only on PowerVM platform
 (02/26) distro_tools.py:DisrtoTool.test;lsslot_pci_-a-7e0e: SKIP: supported only on PowerVM platform
 (03/26) distro_tools.py:DisrtoTool.test;lsslot_phb-a896: SKIP: supported only on PowerVM platform
 (04/26) distro_tools.py:DisrtoTool.test;lsslot_pci_-o-0092: SKIP: supported only on PowerVM platform
 (05/26) distro_tools.py:DisrtoTool.test;lsslot_slot-41a1: SKIP: supported only on PowerVM platform
 (06/26) distro_tools.py:DisrtoTool.test;netstat_a-d781: PASS (2.04 s)
 (07/26) distro_tools.py:DisrtoTool.test;netstat_l-c34c: PASS (1.98 s)
 (08/26) distro_tools.py:DisrtoTool.test;netstat_s-75c2: PASS (1.97 s)
 (09/26) distro_tools.py:DisrtoTool.test;netstat_r-cab9: PASS (2.00 s)
 (10/26) distro_tools.py:DisrtoTool.test;netstat_i-1175: PASS (1.99 s)
 (11/26) distro_tools.py:DisrtoTool.test;lsprop_R-5739: PASS (138.62 s)
 (12/26) distro_tools.py:DisrtoTool.test;lsprop_r-87b2: PASS (84.59 s)
 (13/26) distro_tools.py:DisrtoTool.test;lsvio_s-46ac: PASS (2.05 s)
 (14/26) distro_tools.py:DisrtoTool.test;lsvio_e-546f: PASS (2.06 s)
 (15/26) distro_tools.py:DisrtoTool.test;lsvio_D-14bd: PASS (1.97 s)
 (16/26) distro_tools.py:DisrtoTool.test;lsvio_d-592f: PASS (2.03 s)
 (17/26) distro_tools.py:DisrtoTool.test;test_opt-a9a0: FAIL: lsdevinfo failed (2.11 s)
 (18/26) distro_tools.py:DisrtoTool.test;lsdevinfo_R-c757: FAIL: lsdevinfo failed (2.06 s)
 (19/26) distro_tools.py:DisrtoTool.test;lsdevinfo_c-d3f7: FAIL: lsdevinfo failed (2.04 s)
 (20/26) distro_tools.py:DisrtoTool.test;usysident_identify_l-e695: FAIL: usysident option -P -s identify failed (2.13 s)
 (21/26) distro_tools.py:DisrtoTool.test;usysident_identify_d-2e76: PASS (4.62 s)
 (22/26) distro_tools.py:DisrtoTool.test;usysident_normal_l-7fe7: FAIL: usysident option -P -s normal failed (1.99 s)
 (23/26) distro_tools.py:DisrtoTool.test;usysident_normal_d-2c07: PASS (4.75 s)
 (24/26) distro_tools.py:DisrtoTool.test;usysattn_normal_l-2266: FAIL: usysattn option -P -s normal failed (2.10 s)
 (25/26) distro_tools.py:DisrtoTool.test;ofpathname_l-94b0: FAIL: ofpathname failed (2.16 s)
 (26/26) distro_tools.py:DisrtoTool.test;ofpathname_a-71ea: FAIL: ofpathname failed (2.10 s)
RESULTS    : PASS 13 | ERROR 0 | FAIL 8 | SKIP 5 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 283.43 s
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2020-02-17T12.14-0558a9f/results.html

**Comment by vaishnavibhat** [comment]
On PowerVM

# avocado run ../distro_tools.py -m ../distro_tools.yaml
/usr/lib/python3.6/site-packages/avocado_framework-75.1-py3.6.egg/avocado/plugins/run.py:214: FutureWarning: The following arguments will be changed to boolean soon: sysinfo, output-check, failfast, keep-tmp and ignore-missing-references.
  FutureWarning)
JOB ID     : f63e2ab1bab33c3da933b68bee5217d415116863
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2020-02-17T12.47-f63e2ab/job.log
 (01/26) ../distro_tools.py:DisrtoTool.test;lsslot_pci-a872: PASS (0.78 s)
 (02/26) ../distro_tools.py:DisrtoTool.test;lsslot_pci_-a-d72e: FAIL: Failed to display hot plug slots (0.96 s)
 (03/26) ../distro_tools.py:DisrtoTool.test;lsslot_phb-1712: PASS (0.80 s)
 (04/26) ../distro_tools.py:DisrtoTool.test;lsslot_pci_-o-58fc: PASS (0.79 s)
 (05/26) ../distro_tools.py:DisrtoTool.test;lsslot_slot-e318: PASS (0.76 s)
 (06/26) ../distro_tools.py:DisrtoTool.test;netstat_a-8c13: PASS (1.07 s)
 (07/26) ../distro_tools.py:DisrtoTool.test;netstat_l-81da: PASS (0.86 s)
 (08/26) ../distro_tools.py:DisrtoTool.test;netstat_s-87cb: PASS (0.86 s)
 (09/26) ../distro_tools.py:DisrtoTool.test;netstat_r-128d: PASS (11.40 s)
 (10/26) ../distro_tools.py:DisrtoTool.test;netstat_i-5aef: PASS (0.89 s)
 (11/26) ../distro_tools.py:DisrtoTool.test;lsprop_R-b250: PASS (0.80 s)
 (12/26) ../distro_tools.py:DisrtoTool.test;lsprop_r-a2f8: PASS (0.83 s)
 (13/26) ../distro_tools.py:DisrtoTool.test;lsvio_s-5480: PASS (1.17 s)
 (14/26) ../distro_tools.py:DisrtoTool.test;lsvio_e-18ac: PASS (0.78 s)
 (15/26) ../distro_tools.py:DisrtoTool.test;lsvio_D-9035: PASS (1.32 s)
 (16/26) ../distro_tools.py:DisrtoTool.test;lsvio_d-d736: PASS (0.79 s)
 (17/26) ../distro_tools.py:DisrtoTool.test;test_opt-5ed5: PASS (1.92 s)
 (18/26) ../distro_tools.py:DisrtoTool.test;lsdevinfo_R-2fb0: PASS (1.66 s)
 (19/26) ../distro_tools.py:DisrtoTool.test;lsdevinfo_c-ca5c: PASS (1.68 s)
 (20/26) ../distro_tools.py:DisrtoTool.test;usysident_identify_l-8eef: FAIL: usysident option -P -s identify failed (0.97 s)
 (21/26) ../distro_tools.py:DisrtoTool.test;usysident_identify_d-8240: PASS (1.67 s)
 (22/26) ../distro_tools.py:DisrtoTool.test;usysident_normal_l-d2e9: FAIL: usysident option -P -s normal failed (0.89 s)
 (23/26) ../distro_tools.py:DisrtoTool.test;usysident_normal_d-3713: PASS (1.61 s)
 (24/26) ../distro_tools.py:DisrtoTool.test;usysattn_normal_l-6ee7: FAIL: usysattn option -P -s normal failed (1.03 s)
 (25/26) ../distro_tools.py:DisrtoTool.test;ofpathname_l-b947: FAIL: ofpathname failed (1.09 s)
 (26/26) ../distro_tools.py:DisrtoTool.test;ofpathname_a-97c6: FAIL: ofpathname failed (1.33 s)
RESULTS    : PASS 20 | ERROR 0 | FAIL 6 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 42.47 s
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2020-02-17T12.47-f63e2ab/results.html

[powerVM.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/4212286/powerVM.log)

**Comment by narasimhan-v** [comment]
@manvanthar @abdhaleegit please review

File: io/pci/distro_tools.py [hunk 1]
```diff
@@ -37,6 +37,8 @@ def setUp(self):
         '''
         self.option = self.params.get("test_opt", default='').split(",")
         self.tool = self.params.get("tool", default='')
+        self.interface = self.params.get("interface", default='')
+        self.location_code = self.params.get("location_code", default='')
```
Comment (narasimhan-v) [inline io/pci/distro_tools.py]: @vaishnavibhat This need not be an input. This can be got from avocado util pci.py

Comment (vaishnavibhat) [reply io/pci/distro_tools.py]: @narasimhan-v Can I pass pci address as input and derive the location and interface name from the pci utility ?

**Comment by manvanthar** [comment]
@vaishnavibhat job.log please

File: io/pci/distro_tools.py (hunk 1 above)
Comment (narasimhan-v) [reply io/pci/distro_tools.py]: exactly, yes

File: io/pci/distro_tools.py [hunk 2]
```diff
@@ -74,6 +76,61 @@ def lsprop(self, option):
...
+        cmd = "lsdevinfo -%s" % option
+        result = process.run(cmd, shell=True, ignore_status=True)
+        if result.exit_status != 0:
+            self.fail("lsdevinfo failed")
+
+    def usys(self, tool, option, pci_device):
+        '''
+        run usysident and usysattn
+        '''
+        location_code = pci.get_cfg(pci_device)['YL']
```
Comment (narasimhan-v) [inline io/pci/distro_tools.py]: @vaishnavibhat please use https://github.com/avocado-framework/avocado/blob/master/avocado/utils/pci.py#L175

**Comment by narasimhan-v** [comment]
@manvanthar job.log is uploaded. Please review.

**Comment by manvanthar** [comment]
for usysident,   Failures are for this message "There is no identify indicator at location code"

For this instance can we read through text and then PASS?

**Comment by smruti77** [comment]
it is handled in new PR #1578

**Pull request closed without merging**